go/libraries/doltcore/creds/creds.go

```diff
@@ -177,3 +194,14 @@ func (dc DoltCreds) RPCCreds(audience string) *RPCCreds {
 		RequireTLS: false,
 	}
 }
+
+func (dcp DoltCredsForPass) ToBase64Str() string {
+	return base64.StdEncoding.EncodeToString([]byte(fmt.Sprintf("%s:%s", dcp.Username, dcp.Password)))
+}
+
+func (dc DoltCredsForPass) RPCCreds() *RPCCreds {
```

Comment:
How deep does the upstream dependency on `*RPCCreds` go? I might prefer to just make `DoltCredsForPass` implement `GetRequestMetadata(ctx context.Context, uri ...string)`, for example. I'm not sure the modal behavior on `*RPCCreds` as implemented now is the best, if we can just separate the two functions.